pkg/config/provider.go

```diff
@@ -295,8 +316,34 @@ func NewProvider(schema []byte, prefix string, modulePath string, metadata []byt
 				terraformResource.Schema = terraformResource.SchemaFunc()
 			}
 		}
-		p.Resources[name] = DefaultResource(name, terraformResource, providerMetadata.Resources[name], p.DefaultResourceOptions...)
+
+		var terraformPluginFrameworkResource *fwresource.Resource
+
+		if isPluginFrameworkResource {
+			// TODO: Consider whether to replace the commented out conditional in the next line with an equivalent conditional for plugin framework.
+			if p.TerraformPluginFrameworkProvider == nil /*|| p.TerraformProvider.ResourcesMap[name] == nil */ {
+				panic(errors.Errorf("resource %q is configured to be reconciled without the Terraform CLI"+
+					"but either config.Provider.TerraformProvider is not configured or the Go schema does not exist for the resource", name))
+			}
+
+			for _, resourceFunc := range terraformPluginFrameworkResourceFunctions {
+				resource := resourceFunc()
+
+				resourceTypeNameReq := fwresource.MetadataRequest{
+					ProviderTypeName: name,
+				}
+				resourceTypeNameResp := fwresource.MetadataResponse{}
+				resource.Metadata(ctx, resourceTypeNameReq, &resourceTypeNameResp)
+				if resourceTypeNameResp.TypeName == name {
+					terraformPluginFrameworkResource = &resource
+					break
+				}
+			}
+		}
+
+		p.Resources[name] = DefaultResource(name, terraformResource, terraformPluginFrameworkResource, providerMetadata.Resources[name], p.DefaultResourceOptions...)
 		p.Resources[name].useNoForkClient = isNoFork
```

Comment:
Now that we are introducing TerraformPluginFramework terminology, could we please refactor "noFork" terminology to TerraformPluginSDK throughout the code base for uniformity and clearer understanding?

Reply:
Let's keep renaming the no-fork external client out of scope for this PR and we will follow-up with a new PR after we merge this one. Thank you.